core/src/main/java/com/flipkart/poseidon/datasources/AbstractDataSource.java

```diff
@@ -50,6 +50,10 @@ public AbstractDataSource(LegoSet legoset, Request request) {
         return this.legoset.getDataSourceExecutor().submit(this.legoset.wrapDataSource(dataSource, dataSource.getRequest()));
     }
 
+    protected DataType executeSync(AbstractDataSource dataSource) throws Exception {
+        return this.execute(dataSource).get();
```

Comment:
True. forgot about that.

Generic type might require a change in Lego too, I'll be taking that up